peepholelib/classifier/classifier_base.py

```diff
@@ -69,23 +70,36 @@ def compute_empirical_posteriors(self, **kwargs):
             raise RuntimeError('No fitting dataloader. Please run fit() first.')
 
         verbose = kwargs['verbose'] if 'verbose' in kwargs else False
+        filter_nan = kwargs['filter_nan'] if 'filter_nan' in kwargs else False
+
+        # if mapping:
+        #     invalid = mapping['non-valid']
+        #     print(invalid)
+        # invalid = mapping['non-valid'] if 'mapping' in kwargs else True
+        # print(invalid)
 
         # pre-allocate empirical posteriors
         _empp = torch.zeros(self.nl_class, self.nl_model)
 
         # iterate over _fit_data
+        
         if verbose: print('Computing empirical posterior')
         for batch in tqdm(self._fit_dl, disable=not verbose):
-            data = self.parser(data=batch, **self.parser_kwargs).to(self.device)
+            data, label = self.parser(data=batch, **self.parser_kwargs)
+            data, label = data.to(self.device), label.to(self.device)
             preds = self._classifier.predict(data)
-            labels = batch['label']
-            for p, l in zip(preds, labels):
+            # labels = batch['label']
+            for p, l in zip(preds, label):
                 _empp[int(p), int(l)] += 1
-        
+                        
+        if filter_nan:
+            _empp = _empp[:, :-1]
+       
```

Comment:
I don`t get what is happening here. Lines 94 to 97 are removing the last column of `empp` and not removing `NaN`s. Line 102 removes `NaN`s by substituting them to 0.

Maybe you are trying to handle an application specific case which I am not aware off?